python/pyspark/testing/connectutils.py

```diff
@@ -168,9 +168,21 @@ class ReusedConnectTestCase(unittest.TestCase, SQLTestUtils, PySparkErrorTestUti
     Spark Connect version of :class:`pyspark.testing.sqlutils.ReusedSQLTestCase`.
     """
 
+    @classmethod
+    def conf(cls):
+        """
+        Override this in subclasses to supply a more specific conf
+        """
+        return SparkConf()
+
     @classmethod
     def setUpClass(cls):
-        cls.spark = PySparkSession.builder.appName(cls.__name__).remote("local[4]").getOrCreate()
+        cls.spark = (
+            PySparkSession.builder.config(conf=ReusedConnectTestCase.conf())
```

Comment:
This currently always uses a new `SparkConf`.
```suggestion
            PySparkSession.builder.config(conf=cls.conf())
```